### PR TITLE
PhpdocAnnotationWithoutDotFixer - fix crash on odd character

### PR DESCRIPTION
**Pull request by keradus**
Replaces and fixes #2038

### CODE REVIEW

File: Symfony/CS/Fixer/Symfony/PhpdocAnnotationWithoutDotFixer.php [hunk 1]
```diff
@@ -43,7 +43,12 @@ public function fix(\SplFileInfo $file, $content)
             foreach ($annotations as $annotation) {
                 if ($annotation->getTag()->valid()) {
                     $line = $doc->getLine($annotation->getEnd());
-                    $line->setContent(preg_replace('/[.。](\s+)$/u', '\1', $line->getContent()));
+
+                    preg_match('/^(.*[.。])(\s*)$/', $line->getContent(), $matches);
```
Comment (SpacePossum) [inline Symfony/CS/Fixer/Symfony/PhpdocAnnotationWithoutDotFixer.php]: you don't need the `/u` flag anymore? (out of interest)

File: Symfony/CS/Fixer/Symfony/PhpdocAnnotationWithoutDotFixer.php [hunk 2]
```diff
@@ -43,7 +43,12 @@ public function fix(\SplFileInfo $file, $content)
...
+
+                    if (0 !== count($matches)) {
+                        $line->setContent(mb_substr($matches[1], 0, -1).$matches[2]);
```
Comment (SpacePossum) [inline Symfony/CS/Fixer/Symfony/PhpdocAnnotationWithoutDotFixer.php]: the `mb` package is not installed on a default PHP install, so we should at it to `composer.json` as req.?

Comment (keradus) [inline Symfony/CS/Fixer/Symfony/PhpdocAnnotationWithoutDotFixer.php]: check the newest commit ;P

File: Symfony/CS/Fixer/Symfony/PhpdocAnnotationWithoutDotFixer.php (hunk 1 above)
Comment (keradus) [inline Symfony/CS/Fixer/Symfony/PhpdocAnnotationWithoutDotFixer.php]: new commit ;)

File: Symfony/CS/Fixer/Symfony/PhpdocAnnotationWithoutDotFixer.php (hunk 2 above)
Comment (keradus) [inline Symfony/CS/Fixer/Symfony/PhpdocAnnotationWithoutDotFixer.php]: yeah, it's not working as well ;/

Comment (keradus) [inline Symfony/CS/Fixer/Symfony/PhpdocAnnotationWithoutDotFixer.php]:
```php
PHP Notice:  iconv_strlen(): Detected an illegal character in input string in /home/keradus/github/PHP-CS-Fixer/vendor/symfony/polyfill-mbstring/Mbstring.php on line 338
```

**Comment by keradus** [comment]
I think that is the best we can do. If `preg_replace` will crash there is not much we can do.

Yes, we could add some fancy logic and do everything on very low level, but it's pointless

**Comment by SpacePossum** [comment]
thanks! :+1:

**Pull request closed without merging**